### ISSUE TITLE
Wrong registration order when environment specified

### ISSUE THREAD

**Issue by liri2006** [Milad-Akarie/injectable#244]
When trying to register singletons with environment, registration order is by name instead by creation order:

```dart
abstract class Service {}

@mobileOrDesktopEnv
@Singleton(as: Service)
class ServiceImpl implements Service {}

@singleton
class AppServiceUser {
  final Service _service;
  AppServiceUser(this._service);
}
```

Which generates:

```dart
const String _mobileOrDesktop = 'mobileOrDesktop';
const String _web = 'web';
// ignore_for_file: unnecessary_lambdas
// ignore_for_file: lines_longer_than_80_chars
/// initializes the registration of provided dependencies inside of [GetIt]
_i1.GetIt $initGetIt(_i1.GetIt get,
    {String? environment, _i2.EnvironmentFilter? environmentFilter}) {
  final gh = _i2.GetItHelper(get, environment, environmentFilter);
  gh.singleton<_i5.AppServiceUser>(_i5.AppServiceUser(get<_i5.Service>()));
  gh.singleton<_i5.Service>(_i5.ServiceImpl(), registerFor: {_mobileOrDesktop});
  return get;
}
```

This setup fails during initialization. 
If AppServiceUser is renamed to W_ServiceUser (just to emphasize the ordering issue) then order is:

```dart
const String _mobileOrDesktop = 'mobileOrDesktop';
const String _web = 'web';
// ignore_for_file: unnecessary_lambdas
// ignore_for_file: lines_longer_than_80_chars
/// initializes the registration of provided dependencies inside of [GetIt]
_i1.GetIt $initGetIt(_i1.GetIt get,
    {String? environment, _i2.EnvironmentFilter? environmentFilter}) {
  final gh = _i2.GetItHelper(get, environment, environmentFilter);
  gh.singleton<_i8.Service>(_i8.ServiceImpl(), registerFor: {_mobileOrDesktop});
  gh.singleton<_i8.W_ServiceUser>(_i8.W_ServiceUser(get<_i8.Service>()));
  return get;
}
```

And everything works as expected.

**Pakage versions:**

injectable: 1.5.0
injectable_generator: 1.5.1

**Comment by Milad-Akarie** [comment]
@liri2006 
fixed in injectable_generator v1.5.2
please confirm and close.

**Comment by liri2006** [comment]
@Milad-Akarie Thank you for the extremely quick reaction and the fix! Everything works as expected with the new version.
